R/zchunk_L226.en_distribution.R

```diff
@@ -65,172 +69,358 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
+      # Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(efficiency = round(efficiency, DIGITS_EFFICIENCY))->
+      L226.GlobalTechEff_en
+
+    # Generates L226.GlobalTechCost_en: Costs of global technologies for energy distribution by interpolating values for all model years
+    A26.globaltech_cost %>%
+      gather(year, input.cost, -supplysector, -subsector, -technology, -minicam.non.energy.input) %>%
+      complete(nesting(supplysector, subsector, technology, minicam.non.energy.input), year = c(year, BASE_YEARS, FUTURE_YEARS)) %>%
+      arrange(supplysector, year) %>%
+      group_by(supplysector) %>%
+      mutate (input.cost = approx_fun(as.numeric(year), input.cost)) %>%
+      ungroup() %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(input.cost = round(input.cost, DIGITS_COST)) ->
+      L226.GlobalTechCost_en
+
+    # Generates L226.GlobalTechShrwt_en: Shareweights of global technologies for energy distribution by interpolating values for all model years
+    A26.globaltech_shrwt %>%
+      gather(year, share.weight, -supplysector, -subsector, -technology) %>%
+      complete(nesting(supplysector, subsector, technology), year = c(year, BASE_YEARS, FUTURE_YEARS)) %>%
+      arrange(supplysector, year) %>%
+      group_by(supplysector) %>%
+      mutate (share.weight = approx_fun(as.numeric(year), share.weight)) %>%
+      ungroup() %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L226.GlobalTechShrwt_en
+
+    #2d. Calibration and region-specific data
+    # Electricity ownuse IO coefs - filter down to the base years and append region IDs
+    L126.IO_R_elecownuse_F_Yh %>%
+      ungroup() %>% #this data apparently came into this chunk grouped, which was preventing deletion of grouped columns
+      filter(year %in% c(BASE_YEARS)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join(calibrated_techs, by = c("sector", "fuel")) %>%
+      select(-calibration, -secondary.output) ->
+      L226.IO_R_elecownuse_F_Yh
+
+    #repeat final year's ownuse ratio into future years and append future years to base years (could perhaps be tied to industrial CHP...but also AUTOELEC)
+    L226.IO_R_elecownuse_F_Yh %>%
+      filter(year == max(BASE_YEARS)) %>%
+      repeat_add_columns(tibble("year" = FUTURE_YEARS)) %>%
+      select(-year.x) %>%
+      rename(year = year.y) %>%
+      bind_rows(L226.IO_R_elecownuse_F_Yh, .) ->
```

Comment:
Why `, .`? Does the order of binding matter?

Reply:
I don't believe so, it just seemed intuitive to me to have the future years come after the base years.